src/Makefile

```diff
@@ -227,7 +231,7 @@ export CLUSTER ?= 0
 export GEARS ?= 0
 endif
 
-tests: $(TARGET)
+tests: #$(TARGET)
```

Comment:
@rafie fluke?

Reply:
@danni-m I disabled it because it's annoying to run the build every time you need to go `make test TEST=yada`.

Reply:
@rafie but this breaks the correctness of the makefile, instead we should make sure that the build is cached properly.

Reply:
@danni-m It's a gentle break. The constraint between build and test it IMO weaker than the one between sources and objects. In CI, we always do build before test, and on the desktop, when doing repetitive tests I found it annoying going through an empty build attempt time and again. So we need to decide whether we allow it. I can control it with an env var, but I find it awkward. So it's your call. :-)